mono/utils/mono-merp.c

```diff
@@ -64,6 +64,61 @@ os_version_string (void)
 	return "";
 }
 
+static gboolean
+starts_with (const char *pre, const char *str)
```

Comment:
We know the length of constant strings, so pass it as an argument.
There's actually a macro for this (G_STRING_CONSTANT_AND_LENGTH), and though it's currently broken the fix is easy, I just haven't pushed it to one of my PRs yet. If you want to use the macro I can push the fix to this PR to get it in, or you can just hard-code the length, whatever works.